spec.bs

```diff
@@ -205,6 +202,24 @@ To <dfn>determine eligibility for an associated site</dfn> given a [=site=] |sit
 3. If |index| is greater than or equal to the [=limit for associated sites=], return false.
 4. Return true.
 
+<h2 id="storage-access-integration">Integration with the Storage Access API</h2>
+
+Modify {{Document/requestStorageAccess()}} to insert the following steps before step 13.5 (i.e. before [=requesting permission to use=]):
+
+1. Let |settings| be <var ignore>doc</var>'s [=relevant settings object=].
+1. Let |topLevelSite| be the result of [=obtain a site|obtaining a site=] from |settings|' [=environment/top-level origin=].
+1. Let |embeddedSite| be the result of [=obtain a site|obtaining a site=] from |settings|' [=environment settings object/origin=].
+2. Let |sameSet| be true if |embeddedSite| is [=eligible for same-party membership when embedded within=] |topLevelSite|.
+3. If |sameSet| is true, the user agent may run <var ignore>process permission state</var> with [=permission/granted=]. This step is [=implementation-defined=].
+
+Modify {{Document/requestStorageAccessForOrigin(origin)}} to insert the following steps before step 13.8 (i.e. before [=requesting permission to use=]):
+
+1. Let |settings| be <var ignore>doc</var>'s [=relevant settings object=].
+1. Let |topLevelSite| be the result of [=obtain a site|obtaining a site=] from |settings|' [=environment/top-level origin=].
+1. Let |embeddedSite| be the result of [=obtain a site|obtaining a site=] from |settings|' [=environment settings object/origin=].
+2. Let |sameSet| be true if |embeddedSite| is [=eligible for same-party membership when embedded within=] |topLevelSite|.
```

Comment:
Consider pulling these steps out to an algorithm ("is same-party with its top-level embedder"?) so you don't need to duplicate them between these two monkeypatches.